### PR TITLE
Fix issue used port

### PR DESCRIPTION
**Pull request by FloLey**
# Description

Fix an issue where some tests would always use the same port, which caused failures if the port was already used for something else.

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [ ] Attached issue to pull request
- [x] Changelog entry
- [x] Type annotations are present
- [x] Code is clear and sufficiently documented
- [x] No (preventable) type errors (check using make mypy or make mypy-diff)
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )

### CODE REVIEW

**Review by arnaudsjs — APPROVED** [review]

**Review by arnaudsjs — CHANGES_REQUESTED** [review]
* I did a grep for `Server()` in out test suite and it seems we have a lot more occurrences of this issue than the two instances fixed in this PR. Do you also fix these?
* I think it would be convenient to create a new fixture that 1) Creates the `Server()` instance, 2) Sets the config in place, 3) yields the server object and 4) in its tear down executes the `server.stop()` method. Or something similar if you see a better abstraction. This way we prevent code duplication and we have guarantees that all config is applied consistently everywhere.

**Comment by FloLey** [comment]
@arnaudsjs I think I should not add a new fixture as there are already quit a few that do similar things. I changed my current solution to use one of those existing fixtures. Do you like it this way?

**Comment by arnaudsjs** [comment]
> @arnaudsjs I think I should not add a new fixture as there are already quit a few that do similar things. I changed my current solution to use an existing helper function (not sure why it isn't a fixture?). Do you like it this way? (I found it by looking at test that were still calling `Server() `)

Nice. I didn't know we already had this functionality. The advantage of using a fixture is that you can include teardown code in the fixture as well. But for me it doesn't have to be a fixture.

**Review by Hugo-Inmanta — APPROVED** [review]
Looks good to me :+1:

**Review by arnaudsjs — CHANGES_REQUESTED** [review]
The test case `test_2151_method_header_parameter_in_body` in `tests/test_protocol.py` needs this fix as well, no?

**Review by arnaudsjs — APPROVED** [review]

File: changelogs/unreleased/issue-test-using-used-port.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+description: Fix an issue where some tests would always use the same port, which caused failures if the port was already used for something else.
+change-type: patch
+destination-branches: [master, iso6, iso5, iso4]
```
Comment (FloLey) [inline changelogs/unreleased/issue-test-using-used-port.yml:4]: @sanderr is this right?

Comment (sanderr) [reply changelogs/unreleased/issue-test-using-used-port.yml:4]:
```suggestion
destination-branches: [master, iso6, iso5]
```
See slack.

**Pull request closed without merging**